modules/segmentshortestpaths/cli/segmentshortestpathparser.cpp

```diff
@@ -0,0 +1,145 @@
+            if (!has_only_digits_dots_commas(argv[i])) {
+                std::stringstream message;
+                message << "Invalid destination point provided (" << argv[i]
+                        << "). Should only contain digits dots and commas" << std::flush;
+                throw CommandLineException(message.str().c_str());
+            }
+            destinationPoint = argv[i];
+        } else if (std::strcmp("-sspt", argv[i]) == 0) {
+            ENFORCE_ARGUMENT("-sspt", i)
+            if (std::strcmp(argv[i], "tulip") == 0) {
+                m_stepType = StepType::TULIP;
+            } else if (std::strcmp(argv[i], "metric") == 0) {
+                m_stepType = StepType::METRIC;
+            } else if (std::strcmp(argv[i], "topological") == 0) {
+                m_stepType = StepType::TOPOLOGICAL;
+            } else {
+                throw CommandLineException(std::string("Invalid step type: ") + argv[i]);
+            }
+        }
+    }
+
+    if (originPoint.empty() || destinationPoint.empty()) {
+        throw CommandLineException("Both -sspo and -sspd must be provided");
+    }
+
+    std::stringstream pointsStream;
+    pointsStream << "x,y";
+    pointsStream << "\n" << originPoint;
+    pointsStream << "\n" << destinationPoint;
+    std::vector<Point2f> parsed = EntityParsing::parsePoints(pointsStream, ',');
+    m_originPoint = parsed[0];
+    m_destinationPoint = parsed[1];
+
+    if (m_stepType == StepType::NONE) {
+        throw CommandLineException("Step depth type (-sspt) must be provided");
+    }
+}
+
+void SegmentShortestPathParser::run(const CommandLineParser &clp, IPerformanceSink &perfWriter) const {
+    auto mGraph = dm_runmethods::loadGraph(clp.getFileName().c_str(), perfWriter);
+
+    std::cout << "ok\nSelecting cells... " << std::flush;
+
+    auto graphRegion = mGraph->getRegion();
+
+    if (!graphRegion.contains(m_originPoint)) {
+        throw depthmapX::RuntimeException("Origin point outside of target region");
+    }
+    if (!graphRegion.contains(m_destinationPoint)) {
+        throw depthmapX::RuntimeException("Destination point outside of target region");
+    }
+    QtRegion r(m_originPoint, m_originPoint);
+    mGraph->setCurSel(r, false);
+
+    r = QtRegion(m_destinationPoint, m_destinationPoint);
+    mGraph->setCurSel(r, true);
+
+    std::cout << "ok\nCalculating shortest path... " << std::flush;
+
+    std::unique_ptr<Communicator> comm(new ICommunicator());
+
+#define CONCAT_(x, y) x##y
+#define CONCAT(x, y) CONCAT_(x, y)
+#define DO_TIMED(message, code)                                                                                       \
+    SimpleTimer CONCAT(t_, __LINE__);                                                                                 \
+    code;                                                                                                             \
+    perfWriter.addData(message, CONCAT(t_, __LINE__).getTimeInSeconds());
```

Comment:
I copied those from depthmapXcli/runmethods.cpp. Should they be made global and just called here?